src/dataspaces.jl

```diff
@@ -46,6 +46,12 @@ The default `Dataspace` used for representing a Julia object `data`:
  - arrays: a simple `Dataspace`
  - `nothing` or an `EmptyArray`: a null dataspace
 """
```

Comment:
Should we update the docs here to include the fact that the new default is   a scalar `Dataspace` for struct types?